src/autopas/tuning/utils/Smoothing.cpp

```diff
@@ -119,42 +87,39 @@ double calculateYFitSimple(const std::vector<autopas::Evidence> &points, size_t
 
   double yFitted = 0.;
   for (size_t j = 0; j < projections.size(); ++j) {
-    yFitted += projections[j] * points[j + firstIndex].value;
+    yFitted += projections[j] * static_cast<double>(points[j + firstIndex].value);
   }
 
   return yFitted;
 }
 
 long autopas::smoothing::smoothLastPoint(const std::vector<Evidence> &points, size_t pointsPerEstimation) {
-  // if one or no points are used for smoothing do nothing
-  if (pointsPerEstimation <= 2) {
+  // smoothening requires at least two points
+  if (pointsPerEstimation < 2 or points.size() == 1) {
     return points.back().value;
   }
-  // if there are not enough points to smooth do nothing
-  if (points.size() < 2) {
-    if (points.size() == 1) {
-      return points[0].value;
-    }
+  // special case: there are no points so return 0 instead of points.back().value
+  if (points.empty()) {
     return 0;
   }
 
   // do not try to use more points than there are.
   pointsPerEstimation = std::min(pointsPerEstimation, points.size());
 
-  // since we will only smooth the last point there is no outer loop and indexToFit shall be fixed
-  size_t indexToFit = points.size() - 1;
-  // find neighborhood
+  // get the index of the point to be smoothed (the last point)
+  const auto indexToFit = points.size() - 1;
+  // get the index of the first point in the neighborhood around indexToFit used for smoothening
   const auto firstIndex = indexToFit - pointsPerEstimation + 1;
 
-  // maxDistFromIntervalEdge = max(xi - xFirst, xLast - xi)
-  auto maxDistFromIntervalEdge = std::max(points[indexToFit].iteration - points[firstIndex].iteration, 0ul);
+  // maxDistFromIntervalEdge = xi - xFirst. By distance, we refer to the distance in iteration number.
```

Comment:
```suggestion
  // maxDistFromIntervalEdge = xi - xFirst.
  // By distance, we refer to the distance in iteration number.
```
Just a readability suggestion